src/fides/api/db/system.py

```diff
@@ -142,11 +141,17 @@ async def upsert_cookies(
     privacy_declaration: PrivacyDeclaration,
     system: System,
 ) -> None:
-    """Upsert cookies for the given privacy declaration: retrieve cookies by name/system/privacy declaration
+    """
+    Upsert cookies for the given privacy declaration: retrieve cookies
+    by name/system/privacy declaration.
+
     Remove any existing cookies that aren't specified here.
     """
-    cookie_list: List[CookieSchema] = cookies or []
-    for cookie_data in cookie_list:
+
+    if not cookies:
```

Comment:
@ThomasLaPiana, you've changed the behavior of this function: previously, if cookies were specified as an empty list on an existing declaration, I would remove those cookies from the declaration altogether.  Because you exit early now, there's no opportunity to remove cookies. 

The request and response declarations are not matching because the request is saying "cookies should be an empty list" on this existing declaration and the response is showing the cookies are still there (because of the change to return if not cookies)

Reply:
@pattisdr exactly what I was looking for, thank you! I'll make that fix (er...revert my breaking change)